path-walk.c

```diff
@@ -111,6 +112,23 @@ static int add_children(struct path_walk_context *ctx,
 		if (type == OBJ_TREE)
 			strbuf_addch(&path, '/');
 
+		if (ctx->info->pl) {
+			int dtype;
+			enum pattern_match_result match;
+			match = path_matches_pattern_list(path.buf, path.len,
```

Comment:
I think this needs to be `base_len` instead of `path.len`.

Reply:
`base_len` corresponds to the length of the parent directory name, I think. But I'll figure out how to add tests for `git backfill --sparse` to be sure.

Reply:
Oh, you're right! I totally expected the `pathname` to contain the parent directory, and not end in the `basename`. The documentation of the `path_matches_pattern_list()` function does not help. But other existing callers agree. The implementation is too obtuse and under-documented to make heads or tails of it. You have to search your way back all the way to https://github.com/git-for-windows/git/commit/68492fc73b7387fa72d09545f5743728c5fdb419 to understand that `basename` is passed in only as an optimization, and is expected to point just after the last slash in `pathname` (or to `pathname`, if it contains no slashes).